scrapy/core/downloader/handlers/http11.py

```diff
@@ -31,8 +32,16 @@ def __init__(self, settings):
         self._pool = HTTPConnectionPool(reactor, persistent=True)
         self._pool.maxPersistentPerHost = settings.getint('CONCURRENT_REQUESTS_PER_DOMAIN')
         self._pool._factory.noisy = False
+
+        self._sslMethod = openssl_methods[settings.get('DOWNLOADER_CLIENT_TLS_METHOD')]
         self._contextFactoryClass = load_object(settings['DOWNLOADER_CLIENTCONTEXTFACTORY'])
-        self._contextFactory = self._contextFactoryClass()
+        # try method-aware context factory
+        try:
+            self._contextFactory = self._contextFactoryClass(method=self._sslMethod)
+        except TypeError:
+            # use defaults
+            self._contextFactory = self._contextFactoryClass()
```

Comment:
Good idea. Worth a mention in the settings docs. `if you use your own client factory, make sure it accepts a TLS method string argument (that you can discard)` or similar